weldx/transformations.py

```diff
@@ -1795,6 +1795,24 @@ def add_cs(
                     lcs,
                 )
 
+    def relabel(self, mapping: Dict[str, str]):
+        """Rename one or more nodes of the graph.
+
+        See `networkx.relabel.relabel_nodes` for details.
+        CSM will always be changed inplace.
+
+        Parameters
+        ----------
+        mapping
+             A dictionary mapping with the old node names as keys and new node names
+             labels as values.
+
+        """
+        if len(self.subsystems) > 0:
+            raise NotImplementedError("Cannot relabel nodes on merged systems.")
```

Comment:
tested locally and seems to raise correctly

we can also make this a ""private" function for now until we are sure this doesn't break anything? @vhirtham

Reply:
> tested locally and seems to raise correctly

copy the local test and make it a real one ;) We don't need to think of a good one for now but a quick and dirty test is better than no test :p

can be "public". The exception catches the only problem

Reply:
test added :)